### ISSUE TITLE
Error with npm run test:watch after updating to 3.4 (kinda Solved - but still confused)

### ISSUE THREAD

**Issue by denisflorkin** [mxstbr/react-boilerplate#1491]
RBP 3.4
Node 7.4.0
npm 4.0.5

**Step to reproduce :**

* `git clone  --depth=1 git@github.com:mxstbr/react-boilerplate.git someFolder`
* `cd someFolder`
* `npm run setup`
* `npm run test:watch`

the last one outputs : 
```
2017-01-20 12:17 node[22342] (FSEvents.framework) FSEventStreamStart: register_with_server: ERROR: f2d_register_rpc() => (null) (-22)
2017-01-20 12:17 node[22342] (FSEvents.framework) FSEventStreamStart: register_with_server: ERROR: f2d_register_rpc() => (null) (-22)
2017-01-20 12:17 node[22342] (FSEvents.framework) FSEventStreamStart: register_with_server: ERROR: f2d_register_rpc() => (null) (-22)
events.js:160
      throw er; // Unhandled 'error' event
      ^

Error: Error watching file for changes: EMFILE
    at exports._errnoException (util.js:1022:11)
    at FSEvent.FSWatcher._handle.onchange (fs.js:1283:11)

npm ERR! Darwin 16.3.0
npm ERR! argv "/usr/local/bin/node" "/usr/local/bin/npm" "run" "test:watch"
npm ERR! node v7.4.0
npm ERR! npm  v4.0.5
npm ERR! code ELIFECYCLE
npm ERR! react-boilerplate@3.4.0 test:watch: `cross-env NODE_ENV=test jest --watchAll`
npm ERR! Exit status 1
npm ERR! 
npm ERR! Failed at the react-boilerplate@3.4.0 test:watch script 'cross-env NODE_ENV=test jest --watchAll'.
npm ERR! Make sure you have the latest version of node.js and npm installed.
npm ERR! If you do, this is most likely a problem with the react-boilerplate package,
npm ERR! not with npm itself.
npm ERR! Tell the author that this fails on your system:
npm ERR!     cross-env NODE_ENV=test jest --watchAll
npm ERR! You can get information on how to open an issue for this project with:
npm ERR!     npm bugs react-boilerplate
npm ERR! Or if that isn't available, you can get their info via:
npm ERR!     npm owner ls react-boilerplate
npm ERR! There is likely additional logging output above.

npm ERR! Please include the following file with any support request:
npm ERR!     /Users/denisflorkin/Projects/update-mxstbr-test/npm-debug.log
```

Following the error I found this [issue](https://github.com/facebook/react-native/issues/10088)

After installing watchman formula - which I did not have at all - via brew 
`brew install watchman`

The command npm run test:watch works as excpected.

Did I miss something or Is there an other solution to this issue ? If not, I did not saw anywhere the need to manually install the watchman brew package. Or is it that is supposed to be installed during the `npm run setup` and it somehow didn't?

Anyway everything is working now, but It confused me to have to installed an external dependency manually after following error.

**Comment by mxstbr** [comment]
This is an issue with your local environment and not a general, required step to install and use `react-boilerplate`. Glad you found the solution!

**Comment by tauren** [comment]
@mxstbr I ran into the same problem and had to run `brew install watchman` to solve it. Shouldn't this be documented somewhere?

**Comment by gihrig** [comment]
@tauren 

> Shouldn't this be documented somewhere?

Probably in [`docs/general/gotchas.md`](https://github.com/react-boilerplate/react-boilerplate/blob/master/docs/general/gotchas.md)

There are similar issues that seem to be related to dependency install errors. see: #1351, #1355, #1511, maybe others. These seem to be related to missing dependencies that are only indirectly related to react-boilerplate. Just why this happens on a very few machines is not known.

A possibility is that `npm run setup` or `npm install` has errored in some way. The project provides a `yarn.lock` file that (pretty much) guarantees a working set of dependencies is installed if the `yarn install` command is used to install dependencies. 

Please try the following:
`rm -rf node_modules`
`yarn install` (may have to install yarn: `brew update`; `brew install yarn`)

As @mxstbr points out above

> This is an issue with your local environment and not a general, required step to install and use react-boilerplate

Still if you would like to collect info from this and the referenced issues and open a PR to add to `gotchas.md` that would be welcome.

**Comment by tauren** [comment]
@gihrig It is possible this happened because I was not working with a fresh clone of react-bootstrap.

This situation happened to me while working on an existing project based on a version of react-boilerplate from about 3 months ago. I merged the 3.4 release into my project and then did a bunch of cleanup to get things working. I did run `npm install`, but I did not run `npm run setup` again.

However, I don't think I ever ran `npm run test:watch` before doing the merge, so I couldn't tell you if there was a problem previously or not.

**Comment by gihrig** [comment]
@tauren `npm run setup` and `npm install` are nearly identical. The only significant difference is that `npm run setup` deletes and creates a fresh `.git` local repository folder.

However, if you make changes to dependencies in `package.json` and _don't_ remove `node_modules` or at least run `npm prune` there is a risk of leaving some old dependencies in place. Even then, npm's cache may contain outdated packages. You can clear npm's cache with `npm cache clean`. Personally, I have had a better experience with `npm prune` and `yarn install` whenever I make dependency changes, or when stale dependencies are suspected.
